spectator/data/spectator/functions/spectator.mcfunction

```diff
@@ -1,8 +1,16 @@
+scoreboard players set @s valid 1
+
 execute store result score @s spec_x run data get entity @s Pos[0] 8
 execute store result score @s spec_y run data get entity @s Pos[1] 64
 execute store result score @s spec_z run data get entity @s Pos[2] 8
 execute store result score @s spec_rx run data get entity @s Rotation[0] 64
 execute store result score @s spec_ry run data get entity @s Rotation[1] 64
 execute store result score @s spec_dim run data get entity @s Dimension
 
-gamemode spectator @s
\ No newline at end of file
+#Conditions for cheating (block is fire/lava, or falling)
+execute unless data entity @s {Air:300s} run scoreboard players set @s valid 0
+execute unless data entity @s {OnGround:1b} run scoreboard players set @s valid 0
```

Comment:
This won't let you go into spectator if floating on water. If you passed the Air:300 check this should allow for water

Reply:
and exactly how do you float on water?

Reply:
Pressing spacebar

Reply:
but how do you even enter the command while having spacebar held down? cos when you press T to open chat, you begin to sink

Reply:
You sink slower than you can type, and the command is ran up to 1s later than you entering it. You have the time it takes to sink from feet to head to get into spec, any later will get you denied for Air time (drowning), but the point is being on water should pass the "on ground" check

Reply:
@Pablete1234 Please view commit 0ae9ad4 on 12th march to see if that's what you mean